fix(python): Parse tuples in array accesses

nmote opened the pull request:
This fixes both a matching issue and an autofix issue (see tests).

Fixes #3387

Test plan: Automated tests

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

nmote committed:
fix(python): Parse tuples in array accesses

This fixes both a matching issue and an autofix issue (see tests).

Fixes #3387

Test plan: Automated tests

aryx commented on cli/tests/e2e/snapshots/test_autofix/test_autofix/rulesautofixpython-ranges.yaml-autofixpython-ranges.py-dryrun/results.json:
```diff
@@ -211,6 +211,170 @@
         "line": 5,
         "offset": 49
       }
+    },
```
We should get rid of those results.json. Does someone read them and make sure
they contain the right thing?
Seems like what matters is that the generated code look good (and there's a .py-fixed nice file for that.
The results.json file seems redundant and unreadable anyway).

nmote replied:
I wrote the test and updated the snapshot before applying the fix, and then looked at the diff in the `results.json` file after applying the fix, but I agree that its utility is dubious especially when with autofix you can see exactly what the result of the autofix will be.

nmote commented:
I still have to figure out the failing benchmark test

nmote commented:
I updated the benchmark snapshots with the new findings. It looks like they are legitimate findings that were missed before. I added a new semgrep-core test with a reduced version of one.

emjin commented:
Nice! Love to see it!
